h2o-core/src/main/java/hex/PartialDependence.java

```diff
@@ -179,30 +218,31 @@ public void compute2() {
       int num_cols_1d_2d = _num_1D+_num_2D_pairs;
       _partial_dependence_data = new TwoDimTable[num_cols_1d_2d];
       
+      int column = 0;
       for (int i = 0; i < num_cols_1d_2d; ++i) {  // take care of the 1d pdp first, then 2d pdp
         boolean workingOn1D = (i < _num_1D);
-        final String col = workingOn1D ? _cols[i] : _col_pairs_2dpdp[i - _num_1D][0];
-        final String col2 = workingOn1D ? null : _col_pairs_2dpdp[i - _num_1D][1];
-        Log.debug("Computing partial dependence of model on '" + col + "'.");
+        final String col = workingOn1D ? _cols[column] : _col_pairs_2dpdp[column - _num_1D][0];
+        final String col2 = workingOn1D ? null : _col_pairs_2dpdp[column - _num_1D][1];
+        final int whichPredictorColumn = i % _predictor_columns.length;
+        Log.debug("Computing partial dependence of model on '" + col + "'"+(_targets == null ? "." : " and class "+_targets[whichPredictorColumn]+"."));
         double[] colVals = extractColValues(col, _nbins, fr.vec(col));
         double[] col2Vals = workingOn1D ? null : extractColValues(col2, _nbins, fr.vec(col2));
 
         Futures fs = new Futures();
         int responseLength = workingOn1D ? colVals.length : colVals.length * col2Vals.length;
-        final double meanResponse[] = new double[responseLength];
-        final double stddevResponse[] = new double[responseLength];
-        final double stdErrorOfTheMeanResponse[] = new double[responseLength];
+        final double[] meanResponse = new double[responseLength];
+        final double[] stddevResponse = new double[responseLength];
+        final double[] stdErrorOfTheMeanResponse = new double[responseLength];
 
```

Comment:
yes, those are ugly, aren't they?  Thanks for correcting them.

Reply:
I just don't like my idea shows me what is bad. Then I need to correct it. :D